### PR TITLE
Test for loading cache from conf in different mode

### PR DESCRIPTION
**Pull request by Ostrokrzew**
Signed-off-by: Slawomir Jankowski <slawomir.jankowski@intel.com>

### CODE REVIEW

File: test/functional/tests/initialize/test_startup_init_config.py [hunk 1]
```diff
@@ -102,3 +102,76 @@ def test_cas_startup(cache_mode, filesystem):
...
+
+
+def validate_cache(cache, cache_mode):
+    try:
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            raise Exception
+    except Exception:
+        TestRun.fail(
+            f"Cache did not load successfully - wrong number of caches: {caches_count}."
```
Comment (katlapinka) [inline test/functional/tests/initialize/test_startup_init_config.py:161]: if len() method will raise exception due to none output of  casadm_parser.get_caches(), than caches_count will be not defined and will raise exception in this line.
I suggest executing casadm_parser.get_caches() in the begining of this method and use only it's output later to check if number of caches is OK and also to check if cache mode of returned cache is OK - then you can remove cache parameter from this method.

File: test/functional/tests/initialize/test_startup_init_config.py [hunk 2]
```diff
@@ -102,3 +102,76 @@ def test_cas_startup(cache_mode, filesystem):
...
+        )
+
+    try:
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 1:
+            raise Exception
+    except Exception:
+        TestRun.fail(f"Cache loaded with wrong number of cores: {cores_count}.")
```
Comment (katlapinka) [inline test/functional/tests/initialize/test_startup_init_config.py:169]: same as in previous comment about not defined cores_count

**Review by katlapinka — CHANGES_REQUESTED** [review]

File: test/functional/tests/initialize/test_startup_init_config.py [hunk 3]
```diff
@@ -102,3 +102,71 @@ def test_cas_startup(cache_mode, filesystem):
...
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_mode_pair", [(CacheMode.WT, CacheMode.WB),
+                                              (CacheMode.WB, CacheMode.WA),
+                                              (CacheMode.WA, CacheMode.PT),
+                                              (CacheMode.PT, CacheMode.WO),
+                                              (CacheMode.WO, CacheMode.WT)])
+def test_cas_start_with_changed_mode(cache_mode_pair):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_startup_init_config.py:114]: start => init

File: test/functional/tests/initialize/test_startup_init_config.py [hunk 4]
```diff
@@ -102,3 +102,71 @@ def test_cas_startup(cache_mode, filesystem):
...
+    """
+    title: Check starting cache in other cache mode by starting OpenCAS service from config.
+    description: |
+      Start cache, create config based on running configuration but with another cache mode,
+      restart OpenCAS service and check if cache defined in config file starts properly.
```
Comment (Deixx) [inline test/functional/tests/initialize/test_startup_init_config.py:119]: all mentions of start => initialize, mention --force

File: test/functional/tests/initialize/test_startup_init_config.py [hunk 5]
```diff
@@ -102,3 +102,71 @@ def test_cas_startup(cache_mode, filesystem):
...
+
+    with TestRun.step(
+            f"Create the configuration file with a different cache mode ({cache_mode_pair[1]})"
+    ):
+        init_conf = InitConfig()
+        init_conf.add_cache(cache.cache_id, cache.cache_device, cache_mode_pair[1])
+        init_conf.add_core(cache.cache_id, core.core_id, core.core_device)
+        init_conf.save_config_file()
+
+    with TestRun.step("Re-init and restart OpenCAS service."):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_startup_init_config.py:145]: Reinitialize OpenCAS service with '--force' option.

File: test/functional/tests/initialize/test_startup_init_config.py [hunk 6]
```diff
@@ -102,3 +102,71 @@ def test_cas_startup(cache_mode, filesystem):
...
+        casadm.stop_all_caches()
+        casctl.init(True)
+        casctl.start()
```
Comment (Deixx) [inline test/functional/tests/initialize/test_startup_init_config.py:148]: start() does nothing after init() (except for displaying errors that cache is already used)

**Review by Deixx — CHANGES_REQUESTED** [review]

**Review by Deixx — APPROVED** [review]

**Review by Deixx — APPROVED** [review]

**Review by katlapinka — APPROVED** [review]

**Pull request merged**